File: src/main/java/fi/quanfoxes/types/Uint.java
package fi.quanfoxes.types;

import fi.quanfoxes.lexer.NumberType;

public class Uint extends Number {
    private static final int BYTES = 4;

    public Uint() {
        super(NumberType.UINT32, 32, "uint");
    }

    @Override
    public int getSize() {
        return BYTES;
    }
}

File: Zigzag/Types/Uint.cs
public class Uint : Number
{
	private const int BYTES = 4;

	public Uint() : base(NumberType.UINT32, 32, "uint") { }

	public override int GetReferenceSize()
	{
		return BYTES;
	}

	public override int GetContentSize()
	{
		return BYTES;
	}
}
